src/main/webapp/js/common.js

```diff
@@ -870,3 +870,64 @@ function hideSingleCollapse(e) {
     $(heading).find('a.btn').hide();
 }
 
+/**
+ * Wrapper for Bootbox.js (available at http://bootboxjs.com/)
+ * "Bootbox.js is a small JavaScript library which allows you to create programmatic dialog boxes using
+ *  Bootstrap modals"
+ */
+var BootboxWrapper = {
+
+    /**
+     * Custom alert dialog to replace default alert() function
+     * Required params: titleText and messegeText
+     * Optional params: okButtonText (defaults to "OK")
+     *                  color (defaults to StatusType.INFO)
+     */
+    showModalAlert: function(titleText, messageText, okButtonText, color) {
+        bootbox.dialog({
+            title: titleText,
+            message: messageText,
+            buttons: {
+                okay: {
+                    label: okButtonText || 'OK',
+                    className: 'modal-btn-ok '
+                               + 'btn-' + color || StatusType.INFO
+                }
+            }
+        })
+        // applies bootstrap color to title background
+        .find('.modal-header').addClass('alert-' + color || StatusType.INFO);
+    },
+
+    /**
+     * Custom confirmation dialog to replace default confirm() function
+     * Required params: titleText, messegeText and okCallback
+     * Optional params: cancelCallBack (defaults to null)
+     *                  okButtonText (defaults to "OK")
+     *                  cancelButtonText (defaults to "Cancel")
+     *                  color (defaults to StatusType.INFO)
+     */
+    showModalConfirmation: function(titleText, messageText, okCallback, cancelCallback,
+                                    okButtonText, cancelButtonText, color) {
+        bootbox.dialog({
+            title: titleText,
+            message: messageText,
+            buttons: {
+                cancel: {
+                    label: cancelButtonText || 'Cancel',
+                    className: 'modal-btn-cancel '
+                               + 'btn-default',
+                    callback: cancelCallback || null
+                },
+                ok: {
+                    label: okButtonText || 'OK',
```

Comment:
@wkurniawan07 Since the constants represent default settings, do you think they should be localized to the wrapper (i.e., in the same "namespace")? Because if we are using the constants to pass in default settings, something is wrong.

Actually is this one of my main concerns with the wrapper methods: if we only want to customize a parameter somewhere near the back of the parameter list (e.g., color), we must pass in `null` arguments for all the previous parameters.